src/dist/mod.rs

```diff
@@ -1169,9 +1169,18 @@ pub(crate) async fn dl_v2_manifest(
             Ok(Some((manifest, manifest_hash)))
         }
         Err(any) => {
-            if let Some(RustupError::ChecksumFailed { .. }) = any.downcast_ref::<RustupError>() {
-                // Checksum failed - issue warning to try again later
-                info!("update not yet available, sorry! try again later")
+            if let Some(err @ RustupError::ChecksumFailed { .. }) =
+                any.downcast_ref::<RustupError>()
+            {
+                // Manifest checksum mismatched.
+                // This might indicate a transient state during an update
+                // on the official release server
+                // (see <https://github.com/rust-lang/rustup/issues/3390>)
+                // or a misconfiguration of a third-party release server
+                // (see <https://github.com/rust-lang/rustup/issues/3885>).
+                warn!("{err}");
+                info!("if you are on the official release server, this is most likely due to an update happening right now, please try again later");
```

Comment:
I think the wording of a "on the [..] server" and "on a [..] server" is a little odd. Also should we limit line lengths on these logging lines to split at better boundaries?

Reply:
@djc I've slightly adjusted the approach to smartly print one `info!()` line only depending on the current `dist_root_server`. This has also made the messages easier to read.